lib/lm_docstache/parser.rb

```diff
@@ -47,7 +62,8 @@ class Parser
     #   will be the value that will replace the matched string
     def initialize(document, data, options = {})
       @document = document
-      @data = data.transform_keys(&:to_s)
+      @data = data.transform_keys(&:to_s).select {|e, v| !v.is_a?(Array) }
+      @data_sequential_replacement = data.transform_keys(&:to_s).select {|e, v| v.is_a?(Array) }
```

Comment:
As an alternative solution, you could have used `Enumerable#partition` in here:
```suggestion
      @data_sequential_replacement, @data =
        data.partition { |_, v| v.is_a?(array) }.map(&:to_h).stringify_keys
```